js/controller.js

```diff
@@ -1388,7 +1388,6 @@ OO.plugin('Html5Skin', function(OO, _, $, W) {
       } else {
         this.state.closedCaptionOptions.cueText = null;
       }
-      this.renderSkin();
```

Comment:
This change worries me a bit. This event is necessary in order to display closed captions and it looks like this change would prevent captions from being displayed in some cases. Consider that there are other events calling `this.renderSkin()`, which might give the impression that this particular call to `this.renderSkin()` is not needed. However, if this event fires and no other action causes the skin to be rendered, then it looks like the closed captions would not be displayed. Have we tested whether closed captions are affected by this change?

I also don't fully understand why calling `this.renderSkin()` inside `onClosedCaptionsInfoAvailable` affects the closed captions menu. It seems like we might not be addressing the root cause here. Can you explain why this change fixes the issue?